scraper_atlanta.py

```diff
@@ -0,0 +1,180 @@
+    headers_text = headers_code.find_elements_by_tag_name("span")
+    for header in headers_text:
+        val = header.text.strip()
+        if val not in [None, '']:
+            headers.append(val)
+    headers.append('entrydate')
+    return headers
+
+
+def scrape_content(headers: list, content_rows: list) -> list:
+    """
+    Get the content of the table
+    :param headers: table header
+    :param content_rows: row elements of the table content, usually presented as `tr` tags
+    :return: returns a list of permit object in dictionary type
+    """
+    objs = []
+    for row in content_rows:
+        cells = row.find_elements_by_tag_name('span')
+        vals = []
+        for cell in cells:
+            val = cell.text.strip()
+            vals.append(val)
+        vals.append(datetime.now())
+        obj = dict(zip(headers, vals))
+        objs.append(obj)
+    return objs
+
+
+def click_next(page_row) -> bool:
+    """
+    The function clicks the `next` button located in the lower right conner of the page to get table content
+    in the next page. The function also returns the clickability of the next button to determine if at the last page.
+    :param page_row: the row element contains the page navigation bar in html
+    :return: return if the next button is still clickable
+    """
+    next_button = page_row.find_elements_by_tag_name('td')[TableHTML.CLICK_BUTTON_POSITION.value]
+    try:
+        next_button.find_element_by_tag_name('a').click()
+    except Exception as e:
+        return False
+    else:
+        return True
+
+
+def scraper_atlanta(chrome_path: str,
+                    url='https://aca3.accela.com/Atlanta_Ga/Welcome.aspx',
+                    days_to_scrape=1) -> dict:
+    """
+    The actual scraping process.
+    :param chrome_path: the path in the system where `chromedriver` is stored
+    :param url: the url of the website to be scraped
+    :param days_to_scrape: days prior to today to start scraping. 1 means from yesterday to today, 2 days in total.
+           The value is defaulted to 2 days.
+    :return: The function returns a list of permit object in JSON-like style.
+    """
+    driver = webdriver.Chrome(chrome_path)
+    driver.get(url)
+
+    # Start the browser
+    driver.find_element_by_xpath(Element.BUILDING_TAB_XPATH.value).click()  # click `building` tab
+    element = driver.find_element_by_xpath(Element.DATE_FROM_XPATH.value)  # find `date_from` input
+    element.send_keys((datetime.today()
+                       - timedelta(days=days_to_scrape)).strftime('%m/%d/%Y'))  # enter `date_from`
+    driver.find_element_by_xpath(Element.SEARCH_BUTTON_XPATH.value).click()  # click the `search` button
+
+    table_rows = get_table(driver, Element.TABLE_ID.value)
+    headers = get_headers(table_rows)
+    permits = dict()
+
+    # determine if there are more than 1 page of the results
+    if len(table_rows) < TableHTML.HEADER_AND_ABOVE.value \
```

Comment:
Better way is to use the existence of the elements of `aca_pagination` to determine if there is only one row or not I think.

Reply:
fixed